### ISSUE TITLE
Helm configured differently within microk8s vs outside

### ISSUE THREAD

**Issue by mjperrone** [helm/helm#11424]
## Helm configured differently within microk8s vs outside

TL:DR I'd like to understand better what context/cluster `helm` is talking to.

I am trying to use helm within `microk8s`. I went through the [macos install instructions](https://microk8s.io/docs/install-macos) and added the observability add on and manually added the vector helm repo:

```
microk8s helm repo add vector https://helm.vector.dev
microk8s helm repo update
```

Then when I run `microk8s helm repo list` I see:

```
$ microk8s helm repo list
NAME                    URL                                               
prometheus-community    https://prometheus-community.github.io/helm-charts
grafana                 https://grafana.github.io/helm-charts             
vector                  https://helm.vector.dev                           
```

However, when I use a `brew` installed `helm` directly (without `microk8s`) I see different repos:

```
$ helm repo list
NAME    URL                               
bitnami https://charts.bitnami.com/bitnami
```

This confused me, so I tried setting my `~/.kube/config` as their docs [suggest](https://microk8s.io/docs/working-with-kubectl) for kubectl:

```
microk8s config > ~/.kube/config
```

However this doesn't change the repo list. I thought `helm` is supposed to respect that config?

Is there a way to learn more about what kubernetes cluster `helm` is connecting to or what contexts it is aware of/choosing? Using `--debug` didn't provide any more information. I didn't even know I had another cluster running!

---------
---------

Output of `helm version`: 
```
$ helm version
version.BuildInfo{Version:"v3.10.0", GitCommit:"ce66412a723e4d89555dc67217607c6579ffcb21", GitTreeState:"clean", GoVersion:"go1.19.1"}
$ microk8s helm version
version.BuildInfo{Version:"v3.9.1+unreleased", GitCommit:"e5ad1e2252aafb988fb68910e317e34ac9489fe0", GitTreeState:"clean", GoVersion:"go1.19"}
```

Output of `kubectl version`:

```
$ kubectl version
WARNING: This version information is deprecated and will be replaced with the output from kubectl version --short.  Use --output=yaml|json to get the full version.
Client Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.0", GitCommit:"a866cbe2e5bbaa01cfd5e969aa3e033f3282a8a2", GitTreeState:"clean", BuildDate:"2022-08-23T17:44:59Z", GoVersion:"go1.19", Compiler:"gc", Platform:"darwin/amd64"}
Kustomize Version: v4.5.7
The connection to the server localhost:8080 was refused - did you specify the right host or port?

 $ microk8s kubectl version
WARNING: This version information is deprecated and will be replaced with the output from kubectl version --short.  Use --output=yaml|json to get the full version.
Client Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.0", GitCommit:"a866cbe2e5bbaa01cfd5e969aa3e033f3282a8a2", GitTreeState:"clean", BuildDate:"2022-08-23T17:44:59Z", GoVersion:"go1.19", Compiler:"gc", Platform:"darwin/amd64"}
Kustomize Version: v4.5.7
Server Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.2", GitCommit:"5835544ca568b757a8ecae5c153f317e5736700e", GitTreeState:"clean", BuildDate:"2022-09-29T07:03:08Z", GoVersion:"go1.19", Compiler:"gc", Platform:"linux/amd64"}
```

Cloud Provider/Platform (AKS, GKE, Minikube etc.):  `microk8s`

**Comment by yxxhero** [comment]
@mjperrone please show the content of `$HOME/.config/helm/repositories.yaml`

**Comment by mjperrone** [comment]
```
$ cat $HOME/.config/helm/repositories.yaml 
cat: /Users/mperrone/.config/helm/repositories.yaml: No such file or directory
```

**Comment by joejulian** [comment]
The default home directory for helm is the one asked for above. There's nothing in [brew](https://github.com/Homebrew/homebrew-core/blob/master/Formula/helm.rb) that changes that. Do you have `$HELM_HOME` defined, or are you running as root or does the microk8s command create some chroot?

The `repositories.yaml` is the source of that list, though. It works even without a kubernetes cluster so it has nothing to do with which kubeconfig you're using, if that helps.

**Comment by mjperrone** [comment]
Thanks for the info!
Turns out my use case doesn't actually require kubernetes so I'm going to solve the problem with docker-compose and avoid helm entirely. So I'm not going to invest more time to understand my issues here
